### PR TITLE
[remote-state] Fix remote-state default output when component missing 

### LINKED ISSUE #47: [remote-state] defaults not provided when stack is missing and ignore_errors is true
## Steps to Reproduce

`main.tf`:
```hcl
module "remote_state" {
  source  = "cloudposse/stack-config/yaml//modules/remote-state"
  version = "0.22.1"

  component = "missing"
  ignore_errors = true

  defaults = {
    example = "output"
  }
}

output "example" {
  value       = module.remote_state.outputs
}
```

1. Using above `main.tf`, run `terraform init && terraform apply -auto-approve`
2. Output should be 

```
example = {
  "example" = "output"
}
```

Instead, no output is produced because `module.remote_state.outputs` is `null`.

## Environment (please complete the following information):

Anything that will help us triage the bug will help. Here are some ideas:
 - Terraform version 1.1.6
 - cloudposse/utils v0.17.23

### PR DESCRIPTION
**Pull request by Nuru**
## what
- [remote-state]  Return default results when component backend is missing
- Add separate example and test for the `backend` module

## why
- Restore intended behavior
- The example for the `backend` module can use `atmos.yaml` CLI config, but the `vars`, `settings` and `env` modules can't (the `utils` provider needs to be updated to support that, and these modules are used just in `examples/complete` and not by any other repos/modules)

## references
- Fixes #47
- Supersedes and closes #46

### CODE REVIEW

**Comment by Nuru** [comment]
/test all

Note: tests are failing because of changes to `cloudposse/utils` provider, not changes in this PR:
```
Searched all stack files, but could not find config for the component 'my-vpc' in the stack 'my-stack'.
```

**Review by srhopkins — DISMISSED** [review]
lgtm

**Review by aknysh — DISMISSED** [review]

File: variables.tf [hunk 1]
```diff
@@ -1,11 +1,13 @@
 variable "stack_config_local_path" {
   type        = string
   description = "Path to local stack configs"
+  default     = "examples/complete/stacks"
```
Comment (Nuru) [inline variables.tf:4]: This is really confusing. `stack_config_local_path` should be required unless it is ignored in some situation, in which case it should be documented as such.

File: variables.tf [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 }
 
 variable "stacks" {
   type        = list(string)
   description = "A list of infrastructure stack names"
+  default     = ["my-stack"]
```
Comment (Nuru) [inline variables.tf:10]: This is equally confusing. `stack` should be required unless it is ignored in some situation, in which case it should be documented as such.

File: test/src/examples_backend_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import (
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+// Test the Terraform module in examples/backend using Terratest.
+func TestExamplesBackend(t *testing.T) {
+	t.Parallel()
```
Comment (Nuru) [inline test/src/examples_backend_test.go:11]: Why have `t.Parallel()` if you are going to run the tests all sequentially? Either remove `t.Parallel()` or copy [code from example](https://github.com/cloudposse/terraform-example-module/blob/2b51aedca9fc38f156aec6818d74cd2fbc5101f6/test/src/examples_complete_test.go#L14-L43) to run them in parallel and change the Makefile.

Comment (aknysh) [reply test/src/examples_backend_test.go]: updated

**Review by aknysh — DISMISSED** [review]

**Pull request merged**